### PR TITLE
Added error handling for startup timeout in repl.

### PR DESCRIPTION
**Pull request by tc0312**
Before this fix, if repl didn't transit from STARTING to another state within 30 seconds, session.waitForStateChange() will throw a TimeoutException. Thus repl will not call server's callback. The exception is swallowed by ExecutionService as it's in a Future. repl will keep running.
As server's callback wasn't called, and repl's YARN process is in RUNNING state, the session is stuck in "STARTING" state on server. If user attempts to delete the session, server would stuck.

With this fix, if repl startup times out, repl will quit with a non zero exit code. YARN will restart repl. If all restart attempts fail, repl's YARN state will transit to FAILED. server will see that and delete won't stuck.

### CODE REVIEW

**Comment by tc0312** [comment]
I cannot reproduce the test failure locally. Is that test stable?

**Comment by vanzin** [comment]
Yeah that test fails sometimes here, not sure why yet. I restarted the test run.

File: repl/src/main/scala/com/cloudera/livy/repl/Main.scala [hunk 1]
```diff
@@ -127,20 +127,28 @@ class ScalatraBootstrap extends LifeCycle with Logging {
...
+
+        info(s"Calling $callbackUrl...")
+        val rep = Http(req OK as.String)
+        rep.onFailure {
+          case _ => System.exit(1)
+        }
+
+        Await.result(rep, Duration(10, TimeUnit.SECONDS))
+      } catch {
+        case e: Throwable => {
```
Comment (vanzin) [inline repl/src/main/scala/com/cloudera/livy/repl/Main.scala]: nit: braces are not necessary

**Comment by vanzin** [comment]
LGTM aside from style nit.

**Comment by tc0312** [comment]
Fixed

**Pull request closed without merging**